docs/source/arxiv/metadocs.rst

```diff
@@ -0,0 +1,175 @@
+3.  Testing Sphinx.
+4.  Creating auto-doc stubs.
+5.  Configuring GitHub pages.
+
+Adding Sphinx to the Pipfile
+''''''''''''''''''''''''''''''
+
+Add the following to the Pipfile's ``[dev-packages]`` section::
+
+    [dev-packages]
+    sphinx = "*"
+    sphinx-autodoc-typehints = "*"
+
+Install Sphinx to the pipenv::
+
+    pipenv install --dev
+
+Add the docs/_build directory to the .gitignore::
+
+    docs/_build
+
+Make and change to a docs directory::
+
+    mkdir docs
+    cd docs
+
+Configuring Sphinx::
+
+    Execute "pipenv run sphinx-quickstart" from your project's "docs" directory and use the defaults except:
+        Separate source and build directories: NO
+        Project name: Enter your project name (e.g., arXiv Compilation Service)
+        Author name(s): arXiv-NG Team
+        Project release: 0.1 (or whatever your version number is
+        autodoc: y
+        intersphinx: y
+        coverage: y
+        viewcode: y
+        githubpages: y
+        Create Makefile? y
+
+    git add Makefile conf.py index.rst
+
+    git commit -m "initial sphinx commit"
+
+``git status`` should then show a clean working tree.
+
+Testing Sphinx
+''''''''''''''''
+
+Since the docs necessitate access to the packages to generate documentation, you must run it with the pipenv activated. There are ways to run the commands with ``pipenv run`` but the relative paths can get messed up.
+
+I prefer to use the following method::
+
+    pipenv shell
+    cd docs
+    make html
+    exit
+
+After this, the docs should be created. in the _build folder. Open "docs/_build/index.html" to see what it made.
+
+Creating auto-doc stubs
+'''''''''''''''''''''''''
+For each module you will create an rst file containing the automod declarations.
+
+For example, with arxiv-compiler, I would create a file "compiler.domain.rst".
+
+.. code-block:: none
+    :caption: compiler.domain.rst
+
+    compiler.domain module
+    =========================
+    
+    .. automodule:: compiler.domain
+        :members:
+        :undoc-members:
+        :show-inheritance: 
+
+After creating this file, I add it to the index.rst so it builds.
+
+.. code-block:: none
+    :caption: index.rst
+
+    .. arXiv Compilation Service documentation master file, created by
+    sphinx-quickstart on Mon Oct 15 16:35:18 2018.
+    You can adapt this file completely to your liking, but it should at least
+    contain the root `toctree` directive.
+    
+    Welcome to arXiv Compilation Service's documentation!
+    =====================================================
+    
+    .. toctree::
+    :maxdepth: 2
+    :caption: Contents:
+    
+    compiler.domain.rst
+    
+    
+    
+    Indices and tables
+    ==================
+    
+    * :ref:`genindex`
+    * :ref:`modindex`
+    * :ref:`search`
+
+If you have an issue with ModuleNotFoundError, then the paths in the conf.py need to be changed. Add the following to the top of docs/conf.py::
+
+
+.. code-block:: python
+    :caption: conf.py
```

Comment:
Might want to double check that this indentation is ok. Sometimes Sphinx is picky about 3 vs 4 spaces

Reply:
sphinx is a 3 space weirdness for the `.. directive::` syntax, but then 4 for the block itself. Who knows :confused:

Reply:
The warning was about a preceding `::` which messed with indentation levels. Fixed.